### PR TITLE
feat: link compared frameworks from the /compare pages

### LINKED ISSUE #856: Add outbound links to compared frameworks on /compare pages
## Problem

The framework comparison pages (`/compare/webjs-vs-*`, shipped in #853) name each competitor (Next.js, Lit, Remix 3, Astro, Rails) but never link to their official sites. Adding one tasteful outbound link per page to the competitor's canonical site is a mild positive SEO signal (relevant, authoritative outbound links reinforce topical relevance and E-E-A-T) and, more importantly, makes the comparisons read as honest and well-cited rather than evasive.

## Design / approach

One outbound link per essay, not per mention (repeated links are noise). Two viable placements:

1. **Header eyebrow (preferred).** Turn the "WebJs vs {competitor}" eyebrow on the `[slug]` page into the link. This gives exactly one above-the-fold outbound link per page with full control over `target`/`rel`, keeps the body prose clean, and needs no change to the shared markdown renderer. Implement by adding a `link:` field to each `compare/*.md` frontmatter, threading it through `modules/compare/types.ts` + both queries, and rendering the eyebrow as an anchor.
2. Inline body link on the first prose mention. Rejected as the default because the shared blog markdown renderer (`website/modules/blog/utils/render-post.ts`) emits `<a rel="noopener noreferrer">` with NO `target="_blank"`, so a body link would open in-tab; adding new-tab there would also affect blog links.

Link targets: nextjs.org, lit.dev, remix.run, astro.build, rubyonrails.org. Point the Remix 3 "beta preview" mention at https://remix.run/blog/remix-3-beta-preview specifically. Use **follow** links (these are genuine editorial citations, not paid/untrusted, so no `nofollow`), `target="_blank"`, `rel="noopener noreferrer"`.

## Implementation notes (for the implementing agent)

- Content: `compare/webjs-vs-{nextjs,lit,remix,astro,rails}.md` (repo root). Add a `link:` (competitor URL) frontmatter field to each. Current fields: title, date, slug, description, competitor, tagline, tags, author.
- Types: add `link` to `Comparison` in `website/modules/compare/types.ts`.
- Queries: `website/modules/compare/queries/list-comparisons.server.ts` and `get-comparison.server.ts` both parse frontmatter via `#lib/frontmatter.ts`; add `link: fm.link || ''` to the returned objects.
- Render: `website/app/compare/[slug]/page.ts` renders the eyebrow `WebJs vs ${c.competitor}`; wrap it in `<a href=${c.link} target="_blank" rel="noopener noreferrer">` when `c.link` is set. Optionally also link the index-card eyebrow in `website/app/compare/page.ts`.
- Landmines: the shared renderer (`render-post.ts`) is used by BOTH blog and compare; do not change its link behaviour just for this. Prose capitalizes WebJs as a proper noun; code/URL tokens stay lowercase (existing convention on these pages). No em-dashes / pause-hyphens in any new prose (AGENTS.md invariant 11).
- Invariants: pages/layouts must stay browser-safe; queries are GET server actions (cacheable/public) already.

## Acceptance criteria

- [ ] Each `/compare/<slug>` page has exactly one outbound link to the competitor's official site (follow, `target="_blank"`, `rel="noopener noreferrer"`).
- [ ] Remix 3 page links to the Remix 3 beta preview post.
- [ ] `webjs check` passes; the website boots and every compare route returns 200.
- [ ] Links resolve and open in a new tab.

### PR DESCRIPTION
**Pull request by vivek7405**
Closes #856

## What

Adds one outbound link per comparison to the competitor's official site, so the `/compare` pages cite what they compare (a mild positive SEO signal + credibility/E-E-A-T, and it reads as honest rather than evasive).

## How

- New `link` frontmatter field on each `compare/*.md`, threaded through `modules/compare/types.ts` and both queries.
- The `[slug]` page renders the "WebJs vs <competitor>" header eyebrow as the outbound link (one prominent link per page, above the fold, no body clutter). Index cards are left un-linked to avoid a nested anchor (the whole card is already a link to `/compare/<slug>`).
- Links: nextjs.org, lit.dev, remix.run, astro.build, rubyonrails.org. The Remix 3 page also links its "beta preview" mention to the Remix 3 beta post.
- All outbound links are follow (genuine editorial citations, not paid/untrusted), open in a new tab, and carry `rel="noopener noreferrer"`.
- The shared blog/compare markdown renderer (`render-post.ts`) now opens absolute-URL links in a new tab; internal `/` links still navigate in place. This is why the Remix inline beta link opens in a new tab, and it is a small improvement for blog external links too.

## Verification

- `webjs check`: all checks pass.
- Booted the site: every `/compare/<slug>` renders its outbound eyebrow link with `target="_blank" rel="noopener noreferrer"`; the Remix beta-post inline link is present and new-tab; index cards link only to `/compare/<slug>` (no nested anchors); the blog still boots and its internal links stay same-tab (0 mis-targeted).

## Docs / surfaces

- N/A for framework docs: this is website marketing content only, no `packages/*` public surface changed. The `link` field is a website-local frontmatter convention.

### CODE REVIEW

File: website/app/compare/[slug]/page.ts [hunk 1]
```diff
@@ -44,7 +45,11 @@ export default async function ComparePage({ params }: { params: { slug: string }
       </nav>
 
       <header class="mb-[64px]">
-        <p class="font-mono text-[12px] uppercase tracking-[0.14em] text-accent font-semibold mb-[20px]">WebJs vs ${c.competitor}</p>
+        <p class="font-mono text-[12px] uppercase tracking-[0.14em] text-accent font-semibold mb-[20px]">
+          WebJs vs ${c.link
+            ? html`<a href=${c.link} target="_blank" rel="noopener noreferrer" class="text-accent no-underline hover:underline">${c.competitor}${NEW_TAB}</a>`
```
Comment (vivek7405) [inline website/app/compare/[slug]/page.ts]: This opens a new tab but omits the site's visually-hidden "(opens in a new tab)" cue that lib/links.ts NEW_TAB adds everywhere else, so it is inconsistent with the site's a11y standard (and the layout test that asserts it).

File: website/modules/blog/utils/render-post.ts [hunk 2]
```diff
@@ -22,8 +22,18 @@ function inline(s: string): string {
...
-    `<a href="${u}" class="text-accent no-underline hover:underline" rel="noopener noreferrer">${t}</a>`);
+  out = out.replace(/\[([^\]]+)\]\(([^)]+)\)/g, (_m, t, u) => {
+    // Absolute (off-site) links open in a new tab so the reader keeps the
+    // article; internal links (starting with /) navigate in place. Escape any
+    // `"` in the URL so it cannot break out of the href attribute.
+    const href = u.replace(/"/g, '%22');
+    const external = /^https?:\/\//.test(u);
+    // For a new-tab link, append the same visually-hidden cue the site chrome
+    // uses (lib/links.ts NEW_TAB) so screen readers announce the tab change.
+    const attrs = external ? ' target="_blank" rel="noopener noreferrer"' : '';
```
Comment (vivek7405) [inline website/modules/blog/utils/render-post.ts:37]: Same a11y gap on the renderer's external links: new tab, no screen-reader cue.

File: website/modules/blog/utils/render-post.ts [hunk 3]
```diff
@@ -22,8 +22,18 @@ function inline(s: string): string {
...
     .replace(/>/g, '&gt;');
-  out = out.replace(/\[([^\]]+)\]\(([^)]+)\)/g, (_m, t, u) =>
-    `<a href="${u}" class="text-accent no-underline hover:underline" rel="noopener noreferrer">${t}</a>`);
+  out = out.replace(/\[([^\]]+)\]\(([^)]+)\)/g, (_m, t, u) => {
+    // Absolute (off-site) links open in a new tab so the reader keeps the
+    // article; internal links (starting with /) navigate in place. Escape any
+    // `"` in the URL so it cannot break out of the href attribute.
+    const href = u.replace(/"/g, '%22');
+    const external = /^https?:\/\//.test(u);
+    // For a new-tab link, append the same visually-hidden cue the site chrome
```
Comment (vivek7405) [inline website/modules/blog/utils/render-post.ts:35]: `href="${u}"` interpolates the URL into a double-quoted attribute without escaping `"`; inline() escapes &<> but not quotes, so a URL with a `"` breaks out of the attribute.

**Review by vivek7405 — COMMENTED** [review]
First pass over the outbound-link change. The mechanics are sound (link threaded through the frontmatter, types, and both queries; the eyebrow hole is a correctly-unquoted webjs attribute hole; index cards stay plain text so there are no nested anchors), but three things needed fixing before this is ready:

1. Both new-tab surfaces skipped the site's own a11y convention. The chrome appends a visually-hidden "(opens in a new tab)" cue (lib/links.ts NEW_TAB) on every external link, and there is a layout test asserting it. The compare eyebrow and the renderer's external links now append the same cue.
2. No test shipped for either new behavior. Added SSR/unit coverage for the renderer's external-vs-internal branching (with the internal counterfactual) and the eyebrow link.
3. The renderer interpolates the markdown URL into a double-quoted href without escaping quotes. Pre-existing and the source is trusted repo markdown, but this PR widens that surface, so a `"` in a URL is now percent-escaped so it cannot break out of the attribute.

All three are fixed in the follow-up commit; tests and check are green.

File: website/app/compare/[slug]/page.ts (hunk 1 above)
Comment (vivek7405) [reply website/app/compare/[slug]/page.ts]: Fixed in [`3925b4d8`](https://github.com/webjsdev/webjs/commit/3925b4d8): the eyebrow now appends the NEW_TAB cue.

File: website/modules/blog/utils/render-post.ts (hunk 2 above)
Comment (vivek7405) [reply website/modules/blog/utils/render-post.ts:37]: Fixed in [`3925b4d8`](https://github.com/webjsdev/webjs/commit/3925b4d8): external links now append the same `(opens in a new tab)` cue.

File: website/modules/blog/utils/render-post.ts (hunk 3 above)
Comment (vivek7405) [reply website/modules/blog/utils/render-post.ts:35]: Fixed in [`3925b4d8`](https://github.com/webjsdev/webjs/commit/3925b4d8): the URL's quotes are now percent-escaped before going into the href.

**Review by vivek7405 — COMMENTED** [review]
Second pass, focused on the shared renderer's link edge cases and the test. Two real gaps in the external-vs-internal branch, both fixed; one pre-existing renderer limitation filed as a follow-up; the tests hold up as real counterfactuals.

- The external check was case-sensitive and missed protocol-relative URLs, so `HTTPS://...` and `//host` were treated as in-place internal navs (no new tab, no rel, no cue). Broadened the check to `/^(https?:)?\/\//i` and added an uppercase-scheme test case.
- The link regex truncates a URL at the first `)` (so a Wikipedia-style `..._(language)` link loses its tail). Pre-existing, not introduced here, and out of scope for this PR, so filed as #860.
- Confirmed the internal-link assertion is a genuine counterfactual and the compare-page test is deterministic (getComparison resolves off import.meta.url, not cwd).

Documented the new-tab behavior in the renderer header. Tests and check green.

File: website/modules/blog/utils/render-post.ts [hunk 4]
```diff
@@ -14,16 +14,30 @@
...
-  out = out.replace(/\[([^\]]+)\]\(([^)]+)\)/g, (_m, t, u) =>
-    `<a href="${u}" class="text-accent no-underline hover:underline" rel="noopener noreferrer">${t}</a>`);
+  out = out.replace(/\[([^\]]+)\]\(([^)]+)\)/g, (_m, t, u) => {
+    // Absolute (off-site) links open in a new tab so the reader keeps the
+    // article; internal links (starting with /) navigate in place. Escape any
+    // `"` in the URL so it cannot break out of the href attribute.
+    const href = u.replace(/"/g, '%22');
+    // External = an absolute http(s) URL (scheme is case-insensitive) or a
+    // protocol-relative `//host`. Internal `/path` links stay same-tab.
+    const external = /^(https?:)?\/\//i.test(u);
```
Comment (vivek7405) [inline website/modules/blog/utils/render-post.ts:34]: Case-sensitive and no protocol-relative handling: `HTTPS://host` and `//host` fall into the internal branch, so an off-site link renders with no target/rel/cue.

Comment (vivek7405) [reply website/modules/blog/utils/render-post.ts:34]: Fixed in [`f2c38776`](https://github.com/webjsdev/webjs/commit/f2c38776): external check is now `/^(https?:)?\/\/\/i` (case-insensitive + protocol-relative).

**Review by vivek7405 — COMMENTED** [review]
Third pass, cross-file consistency and completeness. `link` is threaded identically through the type and both queries; the [slug] eyebrow ternary is well-formed (empty string falls to plain text, the link branch nests a single anchor with the NEW_TAB cue); the index cards correctly omit the outbound link to avoid a nested anchor; no invariant-11 punctuation in the new prose; no sitemap/metadata surface needs the link. Nothing left to fix.

**Review by vivek7405 — COMMENTED** [review]
Reviewed the visible-link change (abae1b3e). The eyebrow is back to plain text and the outbound link is now an explicit, underlined "Visit the <competitor> site" link under the tagline with an aria-hidden arrow and the screen-reader cue. Template is well-formed (href is a normal unquoted attribute hole), no nested anchors, no dead code, and the test's label assertion pins the visible link so a revert to hover-only would fail it. Nothing to fix.
